File: src/ax_eval/storage.py
"""Local JSON result storage."""

from __future__ import annotations

import json
from pathlib import Path

from .runner import RunReport


def save_report(report: RunReport, path: str | Path) -> Path:
    p = Path(path)
    p.parent.mkdir(parents=True, exist_ok=True)
    payload = {
        "pack": report.pack,
        "pack_version": report.pack_version,
        "harnesses": report.harnesses,
        "results": [r.to_dict() for r in report.results],
    }
    p.write_text(json.dumps(payload, indent=2))
    return p


def load_report(path: str | Path) -> dict:
    return json.loads(Path(path).read_text())

File: src/storage.ts
/** Local JSON result storage. */
import { mkdirSync, readFileSync, writeFileSync } from "node:fs";
import { dirname } from "node:path";
import type { RunReport } from "./runner.js";

export function saveReport(report: RunReport, path: string): string {
  mkdirSync(dirname(path), { recursive: true });
  writeFileSync(path, JSON.stringify(report, null, 2));
  return path;
}

export function loadReport(path: string): RunReport {
  return JSON.parse(readFileSync(path, "utf8")) as RunReport;
}
